data/api/server-server/user_keys.yaml

```diff
@@ -12,243 +12,221 @@
+                    device ID to algorithm name.
                   additionalProperties:
-                    type: string
-                    description: algorithm
-                    example: "signed_curve25519"
-                example: {
-                  "@alice:example.com": {
-                    "JLAFKJWSCS": "signed_curve25519"
-                  }
-                }
-            required:
-              - one_time_keys
+                    type: object
+                    additionalProperties:
+                      type: string
+                      description: algorithm
+                      example: signed_curve25519
+                  example:
+                    "@alice:example.com":
+                      JLAFKJWSCS: signed_curve25519
+              required:
+                - one_time_keys
+        required: true
       responses:
-        200:
+        "200":
           description: The claimed keys.
-          schema:
-            type: object
-            properties:
-              one_time_keys:
+          content:
+            application/json:
+              schema:
                 type: object
-                description: |-
-                  One-time keys for the queried devices. A map from user ID, to a
-                  map from devices to a map from `<algorithm>:<key_id>` to the key object.
+                properties:
+                  one_time_keys:
+                    type: object
+                    description: |-
+                      One-time keys for the queried devices. A map from user ID, to a
+                      map from devices to a map from `<algorithm>:<key_id>` to the key object.
 
-                  See the [Client-Server Key Algorithms](/client-server-api/#key-algorithms) section for more information on
-                  the Key Object format.
-                additionalProperties:
-                  type: object
-                  additionalProperties:
-                    type:
-                      - string
-                      - type: object
-                        title: KeyObject
-                        properties:
-                          key:
-                            type: string
-                            description: The key, encoded using unpadded base64.
-                          signatures:
-                            type: object
-                            title: Signatures
-                            additionalProperties:
-                              type: object
-                              additionalProperties:
+                      See the [Client-Server Key Algorithms](/client-server-api/#key-algorithms) section for more information on
+                      the Key Object format.
+                    additionalProperties:
+                      type: object
+                      additionalProperties:
+                        oneOf:
+                          - type: string
+                          - type: object
```

Comment:
I don't think it's the proper fix that should be applied. See #1351.

Reply:
As I also commented there - it's almost proper, except that it should be one level of `additionalProperties` deeper.